iml-agent/src/main.rs

```diff
@@ -31,15 +35,33 @@ async fn main() -> Result<()> {
     let registry_keys: Vec<iml_wire_types::PluginName> = registry.keys().cloned().collect();
     let sessions = session::Sessions::new(&registry_keys);
 
-    tokio::spawn(
+    let (reader, reader_reg) = AbortHandle::new_pair();
+    tokio::spawn(Abortable::new(
         reader::create_reader(sessions.clone(), agent_client.clone(), registry)
             .map_err(|e| {
                 tracing::error!("{}", e);
             })
             .map(drop),
-    );
+        reader_reg,
+    ));
+
+    let (poller, poller_reg) = AbortHandle::new_pair();
+    tokio::spawn(Abortable::new(
+        poller::create_poller(agent_client, sessions.clone()),
+        poller_reg,
+    ));
+
+    let mut sigterm = signal(SignalKind::terminate()).expect("Could not listen to SIGTERM");
+    let mut sigint = signal(SignalKind::interrupt()).expect("Could not listen to SIGINT");
+    futures::select! {
```

Comment:
Non-macro way to do this:

`future::select(sigterm.recv().boxed(), sigint.recv().boxed()).await;`

Reply:
Better :)